src/source/source_cache.js

```diff
@@ -165,7 +165,7 @@ class SourceCache extends Evented {
     _tileLoaded(tile, id, previousState, err) {
         if (err) {
             tile.state = 'errored';
-            this._source.fire('error', {tile: tile, error: err});
+            if (err.status !== 404) this._source.fire('error', {tile: tile, error: err});
```

Comment:
The 404 status code strongly implies the "Not found" status text.

 > [At the HTTP level, a 404 response code is followed by a human-readable "reason phrase". The HTTP specification suggests the phrase "Not Found" and many web servers by default issue an HTML page that includes both the 404 code and the "Not Found" phrase.](https://en.wikipedia.org/wiki/HTTP_404)

I cannot imagine any cases in which a custom status message on a 404 response would necessitate different error handling. If we discover one in the future, we can accommodate it  then. 😄